returnDetailAddendumB.go

```diff
@@ -78,17 +78,55 @@ func (rdAddendumB *ReturnDetailAddendumB) Parse(record string) {
 func (rdAddendumB *ReturnDetailAddendumB) UnmarshalJSON(data []byte) error {
 	type Alias ReturnDetailAddendumB
 	aux := struct {
+		// string is used here so we can check if the date is "" and
+		// avoid trying to unmarshal that value into a time.Time
+		PayorBankBusinessDate string `json:"payorBankBusinessDate"`
 		*Alias
 	}{
-		(*Alias)(rdAddendumB),
+		Alias: (*Alias)(rdAddendumB),
 	}
 	if err := json.Unmarshal(data, &aux); err != nil {
 		return err
 	}
+	if aux.PayorBankBusinessDate != "" {
+		if dateErr := rdAddendumB.PayorBankBusinessDate.UnmarshalJSON(
+			[]byte("\""+aux.PayorBankBusinessDate+"\""),
+		); dateErr != nil {
+			return dateErr
+		}
+	} else if !rdAddendumB.PayorBankBusinessDate.IsZero() {
+		// checking if the current value is the zero value and setting it to
+		// the zero value if it isn't
+		rdAddendumB.PayorBankBusinessDate = time.Time{}
+	}
```

Comment:
This solution works well _if_ the date/time passed in is formatted as RFC3339, but the spec calls for the date to be in `YYYYMMDD` format. This suggestion would allow the JSON API to support both RFC3339 _and_ `YYYYMMDD`:

```suggestion
	// The spec calls for this field to be formatted as YYYYMMDD, but we should handle
	// RFC3339 as well for convenience. Default to empty time.Time{} if no date was passed in.
	rdAddendumB.PayorBankBusinessDate = time.Time{}
	if aux.PayorBankBusinessDate != "" {
		parsed, err := time.Parse(time.RFC3339, aux.PayorBankBusinessDate)
		if err != nil {
			parsed, err = time.Parse("20060102", aux.PayorBankBusinessDate)
			if err != nil {
				return err
			}
		}
		rdAddendumB.PayorBankBusinessDate = parsed
	}
```

Reply:
Sounds good to me